SP-1606: Updates to describe the content of DP1 and updates to the AP plans (#41)
lsst/rtn-011: access.tex

\subsection{Rubin Science Platform}

The Rubin Science Platform (RSP), described in \citeds{LSE-319}, is a set of integrated web-based applications, services, and tools that provides access to the Rubin data products and enables next-to-the-data analysis. 
The RSP comprises three different ``Aspects'': a \emph{Portal} Aspect designed to provide an environment for data discovery, query, filtering, and visualization, a \emph{Notebook} Aspect to enable next-to-the-data analysis, and an \emph{API} Aspect for programmatic access to the Rubin data products via Virtual Observatory (VO) interfaces.
The Portal and Notebook Aspects of the RSP make use of the same APIs as the API Aspect to internally access the LSST datasets.

The RSP is currently under active development and a fully functional RSP is not expected to be available until DR1. 
An early preview of the RSP is deployed at the US DAC and will be used to host the Early Science datasets.
New functionality will be deployed incrementally, as it becomes available. 

The following functionality is already deployed and operational on the RSP:
\begin{itemize}

> \item TAP and Butler access to catalogs and images;

Reviewer: @gpdf do we want to mention the limited write support for butler given that this is worse than what people had with dp0.2?

Author: Yes - good catch. I will add it in

Author: My understanding is that there is no writeable Butler for DP1?

Reviewer: See discussion in https://dmtn-303.lsst.io/v/DM-48071/index.html where we are talking about local sqlite for writes and some tooling to help with that. (although that tech note hasn't distinguished DP1 facilities from DR1).

Author: I added a comment on the Butler being read only for DP1 -- but it is not clear to me what we are advising users to do beyond save their analysis data products to their local space.

Reviewer: We would hope to have some tooling to make that saving to local space a bit easier than them having to know how to save the python object to disk. The tech note talks about tooling to simplify setting up a local sqlite butler that they can `butler.put` to but it has still not been approved so the actual scope of the work is as yet undefined. cc/ @TallJimbo

Author: I don't want to mention anything in RTN-011 unless we are confident we will provide it. I'll add a comment that more details will come closer to the release date.

Author: After thinking about this  - I would prefer not to say anything so specific/technical about any part of the system in this document.  RTN-011 is a high-level strategy for enabling early science aimed at the community to help them prepare and support proposals. Such technical details are better in the online technical documentation, e.g. dp1.lsst.io

Reviewer 2: I agree with this strategy; this isn't the place to go into specifics on something that's actually fairly difficult to explain.

What we can say at this level is just that users will be able to access and run the pipelines code in their notebooks.

> \item TAP and Butler access to catalogs and images;

Reviewer 2: ```suggestion
\item TAP and Butler access to catalogs and images;
\item DataLink annotations in TAP query results for access to light curves and other related information;
```

\item ObsCore data model and ObsTAP service for image metadata and searches;
\item Image retrieval via https;

> \item IVOA SODA service for single image cutouts, one at a time;

Reviewer 2: ```suggestion
\item IVOA SODA service for cutouts from individual single-epoch images and coadded image tiles;
```

\item DataLink annotations to the ObsTAP service for access to the SODA service;

> \item Authenticated HiPS data service.

Reviewer 2: ```suggestion
\item Authenticated HiPS data service for seamless pan-and-zoom access to coadded data.
```

\end{itemize}

The following is a summary of new RSP functionality planned to be available with  DP1: 
\begin{itemize}
\item IVOA-compatible SIA image service;
\item Qserv query temporary uploads;
\item User query history capabilities;
\item Context-aware documentation, e.g pop-ups in the portal, documentation in-context such as  "click on the column name and go to the page that explains it in detail";
\item Some Portal-Notebook integration features such as  seeding a notebook with a query that was executed in the Portal. 
\end{itemize}

RSP functionality not yet available but that is expected by DR1:
\begin{itemize}

> \item Bulk cutout service;

Reviewer 2: ```suggestion
\item Bulk cutout services, both for individual targets across a range of epochs, and for lists of multiple targets;
```

The RSP comprises three different ``Aspects'': a \emph{Portal} Aspect designed to provide an environment for data discovery, query, filtering, and visualization, a \emph{Notebook} Aspect to enable next-to-the-data analysis, and an \emph{API} Aspect for programmatic access to the Rubin data products via Virtual Observatory (VO) interfaces.
The Portal and Notebook Aspects of the RSP make use of the same APIs as the API Aspect to internally access the LSST datasets.

The RSP is currently under active development and a fully functional RSP is not expected to be available until DR1. 
An early preview of the RSP is deployed at the US DAC and will be used to host the Early Science datasets.
New functionality will be deployed incrementally, as it becomes available. 

The following functionality is already deployed and operational on the RSP:
\begin{itemize}
\item TAP and Butler access to catalogs and images;
\item ObsCore data model and ObsTAP service for image metadata and searches;
\item Image retrieval via https;

> \item IVOA SODA service for single image cutouts, one at a time;

Author: Updated the text and added a footnote

> \item IVOA SODA service for single image cutouts, one at a time;

Reviewer 2: It refers to any of the served processed images.  So in the baseline of DP1 that means single-visit calexps and coadd tiles.  If difference images are added to DP1 it would also work for them.

> \item IVOA SODA service for single image cutouts, one at a time;

Reviewer 2: It's a stretch goal to support mosaiced cutouts across tile boundaries, but as a stretch goal I wouldn't mention it here.

\item DataLink annotations to the ObsTAP service for access to the SODA service;
\item Authenticated HiPS data service.
\end{itemize}

There will be an important change to the Butler for DP1. 
To date, science users have accessed Rubin data products on the RSP using a  full read-write \textit{Direct Butler}, which connects to an SQL database.
From DP1, for both scaling and security reasons, science users will access the  Rubin data products  via a new \textit{Remote Butler}, which interacts with a server via a REST API. 
The  \textit{Remote Butler} read and query API will be the same as for the DP0-era \textit{Direct Butler}, however the new \textit{Remote Butler} will not  support user write operations.  
Users will be able to write the outputs of their analyses to local space, and Rubin anticipates providing some tooling to support local write. 

> Various options. 

Author: This section has all been reviewed and updated

Users will also be able to set up a local sqlite Butler to which they can write the outputs of their analysis.  
More details on setting up a local writeable Butler can be found in \citeds{dmtn-303}.

The following is a summary of new RSP functionality expected for DP1: 
\begin{itemize}
\item IVOA-compatible SIA image service;
\item Qserv query temporary uploads;
\item User query history capabilities;

> \item Context-aware documentation, e.g pop-ups in the portal, documentation in-context such as  "click on the column name and go to the page that explains it in detail";

Reviewer 2: also "e.g.," please

\item Some Portal-Notebook integration features such as  seeding a notebook with a query that was executed in the Portal. 
\end{itemize}

RSP functionality not yet available but that is expected by DR1:
\begin{itemize}
\item Bulk cutout service;
\item PSF retrieval service;
\item Data product recreation service; 
\item Parallel computing;
\item Batch processing; 
\item Support for collaborative work;
\item WebDAV service to edit files on their RSP from their preferred device. 
\item Dask for parallel computing;
\end{itemize} 

RSP functionality that is under consideration for post-DR1:
\begin{itemize}
\item Access to GPUs;
\item Bringing individual resources to the RSP, e.g. additional compute paid for by indoviduals. 
\end{itemize} 

\subsection{Community Brokers }
Alerts are fully world-public and will be accessible via one or more of the nine Rubin-endorsed Community Brokers\footnote{See \url{https://www.lsst.org/scientists/alert-brokers}}.
During the commissioning period, Rubin will work with the Community Brokers to integrate them \citedsp{rtn-010}.
Community access to early alerts will depend on the readiness of the Community Brokers. 

> At this stage, we are expecting the first alerts from commissioning to become available via Community Brokers sometime following System First Light \citeds{RTN-061}. 

Author: No we cannot . Work is ongoing on the PPDB  and I would rather not set any expectations in advance of that work. I anticipate issuing another major release of RTN-011 in advance of the first alerts.

Alerts are fully world-public and will be accessible via one or more of the nine Rubin-endorsed Community Brokers\footnote{See \url{https://www.lsst.org/scientists/alert-brokers}}.
During the commissioning period, Rubin will work with the Community Brokers to integrate them \citedsp{rtn-010}.
Community access to early alerts will depend on the readiness of the Community Brokers. 

> At this stage, we are expecting the first alerts from commissioning to become available via Community Brokers sometime following System First Light \citeds{RTN-061}. 

Author: Updated to now say System First Light
